kubernetes/RELEASES.md

```diff
@@ -0,0 +1,127 @@
+# Deploy to Maven release process
+
+This document describes how to configure and use the [Maven release
+plugin](http://maven.apache.org/maven-release/maven-release-plugin) to publish
+to [Sonatype](http://central.sonatype.org/).
+
+Releases are done on an as-needed basis, and this doc applies only to
+[OWNERS](https://github.com/kubernetes-client/java/blob/master/OWNERS).
+
+This does _not_ describe the process of cherry-picking changes onto release
+branches.
+
+## One time setup
+
+You will need to have the following in place:
+
+1. SSH keys for an account capable of pushing commits & tags to
+   https://github.com/kubernetes-client/java. These will be used by the release
+   plugin to push an updated pom.xml along with a tag corresponding to the
+   release being performed. If you don't have these keys, follow [this
+   guide](https://help.github.com/articles/connecting-to-github-with-ssh/).
+
+2. [A Sonatype JIRA
+   account](https://issues.sonatype.org/secure/Signup!default.jspa) that's been
+   authorized to publish to `io.kubernetes:client-java`. With your credentials
+   in-hand, place these in your `settings.xml` (typically
+   `~/.m2/settings.xml`) config file. An example config is:
+   ```xml
+<settings>
+  <servers>
+    <server>
+      <id>ossrh</id>
+      <username>your-jira-id</username>
+      <password>your-jira-pwd</password>
+    </server>
+  </servers>
+</settings>
+   ```
+
+3. A GPG key (specifics TBD) __TODO(lwander)__
+
+## Doing a release
+
+There are three stages to a release explained in detail below:
+
+### 1. Collect requirements and propose your release
+
+Prior to publishing a release, you need to collect three release-specific pieces
+of information:
+
+1. This release's version. We follow [semver](http://semver.org/) to determine
+   release versions.
+
+2. This release's changelog. This can generally be inferred from the commit
+   history, which may need some modification to make presentable. A quick way
+   to collect the commit history is to run
+   `git log <last-release-tag>..HEAD --online`.
+
+3. This release's Kubernetes API compatability changes (if applicable).
+   Generally, if this release reduces functionality with a specific version of
+   the Kubernetes API it's worth bumping the major version number to indicate a
+   breaking change.
+
+File an [issue](https://github.com/kubernetes-client/java/issues) with the
+title `Propose release <VERSION>`, and open a PR against the `master` branch
+with an updated changelog and compatibility matrix (if necessary). Once all
```

Comment:
I think 'all' is overkill here and make release process slow. We can say at least one (or even two when we had more owners).